types/block.go

```diff
@@ -56,6 +56,20 @@ func CalculateCoinbase(height BlockHeight) Currency {
 	return NewCurrency64(base).Mul(SiacoinPrecision)
 }
 
+// CalculateNumSiacoins calculates the number of siacoins in circulation at a
+// given height.
+func CalculateNumSiacoins(height BlockHeight) Currency {
+	deflationBlocks := BlockHeight(InitialCoinbase - MinimumCoinbase)
+	avgDeflationSiacoins := CalculateCoinbase(0).Add(CalculateCoinbase(height)).Div(NewCurrency64(2))
+	if height <= deflationBlocks {
+		deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(height + 1)))
+		return deflationSiacoins
+	}
+	deflationSiacoins := avgDeflationSiacoins.Mul(NewCurrency64(uint64(deflationBlocks + 1)))
+	trailingSiacoins := NewCurrency64(uint64(height - deflationBlocks)).Mul(CalculateCoinbase(height))
+	return deflationSiacoins.Add(trailingSiacoins)
```

Comment:
half a second is a pretty big deal when you are going to be performing that operation many times, especially in debug mode that operation is going to get run every block. Today, that's 300 million total bignum operations (an average of 12,500 per block for 25,000 blocks), or about 2.5 minutes.

I think this is definitely a place where the optimization is worth the tradeoff.